### ISSUE TITLE
MM_ParallelGlobalGC::workThreadGarbageCollect Undefined

### ISSUE THREAD

**Issue by dnakamura** [eclipse/omr#1541]
MM_ParallelGlobalGC::workThreadGarbageCollect is referenced: https://github.com/eclipse/omr/blob/1a3c565552941bea137b741351d506da61211d66/gc/base/standard/ParallelGlobalGCTask.cpp#L25
 but never defined

**Comment by ktbriggs** [comment]
Yes it is (defined). 

From omr.gc/base/standard/ParallelGlobalGCTask.hpp
Lines 36-39 in c9e2a14
```
class MM_ParallelGlobalGCTask : public MM_ParallelTask
{
private:
	MM_ParallelGlobalGC *_collector;
```
From omr.gc/base/standard/ParallelGlobalGC.hpp
Line 269 in 3519c5d
```
	void workThreadGarbageCollect(MM_EnvironmentBase *env);
```
workThreadGarbageCollect(...) is a public member of MM_ParallelGlobalGC.

**Comment by charliegracie** [comment]
@ktbriggs There is a definition in the .hpp but it appears the actual implementation is missing.

**Comment by ktbriggs** [comment]
@charliegracie Right. There is a signature for workThreadGarbageCollect(...) in ParallelGlobalGC.hpp and no implementation in ParallelGlobalGC.cpp. This has been the case since initial contribution of these files along with the rest of OMR (bc344ed) over a year ago. 

This is obviously not correct, but can I ask why (or how -- if it is all of a sudden breaking builds) it is appearing as a problem only now?

**Comment by dnakamura** [comment]
Its not explicitly breaking anything,  but as part of CMake work I was linking with `-Wl,-z,defs`  to quickly catch issues where i was forgetting to link things

**Comment by charliegracie** [comment]
Do we use MM_ParallelGlobalGCTask for global collections?  If we do then how does the GC even work?

**Comment by ktbriggs** [comment]
MM_ParallelGlobalGC uses distinct tasks for mark, sweep, compact. MM_ParallelGlobalGCTask has been around from some time but is never instantiated or used (which is probably why linker has not complained before). So ParallelGlobalGCTask.[ch]pp can be deleted, along with the dangling workThreadGarbageCollect() method declaration ParallelGlobalGC.hpp.

**Comment by charliegracie** [comment]
This was resolved in #1547
